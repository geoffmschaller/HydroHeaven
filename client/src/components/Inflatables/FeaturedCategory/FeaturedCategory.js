import React from 'react';
import styles from './FeaturedCategory.module.css';
import HOT_TUB_IMAGE from '../../../static/images/categories/hot_tub_spas.jpg';
import SWIM_SPAS_IMAGE from '../../../static/images/categories/swim_spas.jpg';
import BBQ_ISLANDS_IMAGE from '../../../static/images/categories/bbq_islands.jpg';
import {Link} from "react-router-dom";
import SectionTitle from "../SectionTitle/SectionTitle";
import VerticalSpacer from "../VerticalSpacer/VerticalSpacer";
import COMMERCIAL_VIDEO from "../../../static/video/hh_commercial.mp4";

const FeaturedCategory = (props) => {

	return (
		<div className={styles.featured}>

			<div className="widthRestriction">
				<SectionTitle title={"View By Category"} darkMode={true}/>
				<VerticalSpacer height={20}/>
				<div className={styles.category}>
					<img src={HOT_TUB_IMAGE} alt=""/>
					<div className={styles.title}>Hot Tubs & Spas</div>
					<p>We searched the country and found the spa brands that provide the highest product quality, affordable pricing, and highest customer
						satisfaction.</p>
					<Link to={"/spas-hot-tubs"}>
						<button>View Hot Tubs</button>
					</Link>
					<div className="clear"/>
				</div>
				<div className={styles.category}>
					<img src={SWIM_SPAS_IMAGE} alt=""/>
					<div className={styles.title}>Swim Spas</div>
					<p>Spas aren't just for relaxing anymore. Get your work out in with one of American Whirlpools award winning swim spas.</p>
					<Link to={"/swim-spas"}>
						<button>View Swim Spas</button>
					</Link>
					<div className="clear"/>
				</div>
				<div className={styles.category}>
					<img src={BBQ_ISLANDS_IMAGE} alt=""/>
					<div className={styles.title}>BBQ Islands</div>
					<p>With its superior products, facilities, and team, Bull Outdoor Products Inc. pioneered the concept of outdoor barbecue islands.</p>
					<Link to={"/bbq-islands"}>
						<button>View BBQ Islands</button>
					</Link>
					<div className="clear"/>
				</div>
				<div className="clear"/>
				<VerticalSpacer height={75}/>
				<SectionTitle title={"Our Sales & Specials"} darkMode={true}/>
				<video src={COMMERCIAL_VIDEO} autoPlay={false} muted={false} controls={true}/>
			</div>

			<div className="clear"/>
		</div>
	);

};

export default FeaturedCategory;