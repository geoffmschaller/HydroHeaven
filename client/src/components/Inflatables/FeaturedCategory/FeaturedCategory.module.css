.featured {
  width: 100%;
  padding: 70px 0;
  background: #297db5;
  background: -moz-radial-gradient(circle, #297db5 0%, #154360 100%);
  background: -webkit-radial-gradient(circle, #297db5 0%, #154360 100%);
  background: -o-radial-gradient(circle, #297db5 0%, #154360 100%);
  background: radial-gradient(circle, #297db5 0%, #154360 100%);
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#297db5",endColorstr="#154360",GradientType=1); }
  .featured .category {
    width: 33%;
    float: left;
    padding: 0 10px; }
    .featured .category img {
      width: 100%;
      margin-bottom: 10px; }
    .featured .category .title {
      font-size: 18px;
      color: #ebf5fb;
      text-transform: uppercase; }
    .featured .category p {
      height: 100px;
      font-size: 15px;
      margin-bottom: 20px;
      color: #abb2b9; }
    .featured .category button {
      color: #fad7a0;
      border: 2px solid #fad7a0;
      float: right; }
      .featured .category button:hover {
        color: white;
        background-color: #fad7a0; }
  .featured video {
    width: 100%;
    border: 3px solid #7fb3d5; }

@media (max-width: 400px) {
  .featured {
    display: block; }
    .featured .category {
      width: 100%;
      margin-bottom: 30px; } }
