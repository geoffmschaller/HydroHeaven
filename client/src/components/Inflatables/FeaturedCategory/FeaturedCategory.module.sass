$name-color: #ebf5fb
$description-color: #abb2b9
$button-color: #fad7a0

.featured
	width: 100%
	padding: 70px 0
	background: rgb(41,125,181)
	background: -moz-radial-gradient(circle, rgba(41,125,181,1) 0%, rgba(21,67,96,1) 100%)
	background: -webkit-radial-gradient(circle, rgba(41,125,181,1) 0%, rgba(21,67,96,1) 100%)
	background: radial-gradient(circle, rgba(41,125,181,1) 0%, rgba(21,67,96,1) 100%)
	filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#297db5",endColorstr="#154360",GradientType=1)

	.category
		width: 33%
		float: left
		padding: 0 10px

		img
			width: 100%
			margin-bottom: 10px

		.title
			font-size: 18px
			color: $name-color
			text-transform: uppercase

		p
			height: 100px
			font-size: 15px
			margin-bottom: 20px
			color: $description-color

		button
			color: $button-color
			border: 2px solid $button-color
			float: right

			&:hover
				color: white
				background-color: $button-color

	video
		width: 100%
		border: 3px solid #7fb3d5

@media(max-width: 400px)
	.featured
		display: block

		.category
			width: 100%
			margin-bottom: 30px