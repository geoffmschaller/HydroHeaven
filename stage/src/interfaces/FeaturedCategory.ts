interface FeaturedCategory {
	image: string,
	name: string,
	description: string,
	link: string
}

export default FeaturedCategory;