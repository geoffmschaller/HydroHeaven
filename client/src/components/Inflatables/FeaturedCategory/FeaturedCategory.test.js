import React from 'react';
import FeaturedCategory from "./FeaturedCategory";

const wrapper = <FeaturedCategory/>;

it("Featured Category Inflatable", () => {
	expect(wrapper).toMatchSnapshot();
});