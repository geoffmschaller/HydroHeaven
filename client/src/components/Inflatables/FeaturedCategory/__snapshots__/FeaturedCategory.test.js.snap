// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured Category Inflatable 1`] = `<FeaturedCategory />`;
